src/Core/Webservice/WebserviceCanBeEnabledConfigurationChecker.php

```diff
@@ -0,0 +1,171 @@
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Translation\TranslatorInterface;
+use RuntimeException;
+
+/**
+ * Looks at server configuration in order to check PrestaShop Webservice can be enabled
+ */
+class WebserviceCanBeEnabledConfigurationChecker
+{
+    const ISSUE_NOT_APACHE_SERVER = 'not_apache_server';
+    const ISSUE_CANNOT_CHECK_APACHE_MODULES = 'cannot_check_apache_modules';
+    const ISSUE_APACHE_MOD_AUTH_BASIC_NOT_AVAILABLE = 'issue_apache_mod_auth_basic_not_available';
+    const ISSUE_APACHE_MOD_AUTH_REWRITE_NOT_AVAILABLE = 'issue_apache_mod_auth_rewrite_not_available';
+    const ISSUE_EXT_SIMPLEXML_NOT_AVAILABLE = 'issue_ext_simplexml_not_available';
+    const ISSUE_HTTPS_NOT_AVAILABLE = 'issue_https_not_available';
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param Configuration $configuration
+     */
+    public function __construct(TranslatorInterface $translator, Configuration $configuration)
+    {
+        $this->translator = $translator;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * Analyses the server configuration (apache configuration and php settings)
+     * to check whether PrestaShop Webservice can be used or not.
+     *
+     * @param Request $request optional
+     *
+     * @return array empty if no errors
+     */
+    public function getErrors(Request $request = null)
+    {
+        $issues = $this->lookForIssues($request);
+
+        $allWarningMessages = $this->getWarningMessages();
+        $selectedWarningMessages = [];
+
+        foreach ($issues as $issue) {
+            if (false === array_key_exists($issue, $allWarningMessages)) {
+                throw new RuntimeException(sprintf('Unexpected configuration issue %s', $issue));
+            }
+
+            $selectedWarningMessages[] = $allWarningMessages[$issue];
+        }
+
+        return $selectedWarningMessages;
+    }
+
+    /**
+     * @param Request $request optional
+     *
+     * @return string[]
+     */
+    private function lookForIssues(Request $request = null)
+    {
+        $issues = [];
+
+        if ($request !== null) {
+            if (strpos($request->server->get('SERVER_SOFTWARE'), 'Apache') === false) {
+                $issues[] = self::ISSUE_NOT_APACHE_SERVER;
+            }
+        }
+
+        if (function_exists('apache_get_modules')) {
+            $apache_modules = apache_get_modules();
+
+            if (false === in_array('mod_auth_basic', $apache_modules)) {
+                $issues[] = self::ISSUE_APACHE_MOD_AUTH_BASIC_NOT_AVAILABLE;
+            }
+
+            if (false === in_array('mod_rewrite', $apache_modules)) {
+                $issues[] = self::ISSUE_APACHE_MOD_AUTH_REWRITE_NOT_AVAILABLE;
```

Comment:
Weird, where we are under a Symfony page ^^ This one can be removed maybe?

Reply:
we try to be iso functional with legacy: are you 100% sure that in Symfony page this condition is not required?